cmd/signature-v4_test.go

```diff
@@ -221,6 +227,63 @@ func TestDoesPresignedSignatureMatch(t *testing.T) {
 			region:   serverConfig.GetRegion(),
 			expected: ErrSignatureDoesNotMatch,
 		},
+		// (8) Should error if the request is not ready yet, ie X-Amz-Date is in the future.
+		{
+			queryParams: map[string]string{
+				"X-Amz-Algorithm":      signV4Algorithm,
+				"X-Amz-Date":           now.Add(1 * time.Hour).Format(iso8601Format),
+				"X-Amz-Expires":        "60",
+				"X-Amz-Signature":      "badsignature",
+				"X-Amz-SignedHeaders":  "host;x-amz-content-sha256;x-amz-date",
+				"X-Amz-Credential":     fmt.Sprintf(credentialTemplate, serverConfig.GetCredential().AccessKeyID, now.Format(yyyymmdd), serverConfig.GetRegion()),
+				"X-Amz-Content-Sha256": payload,
```

Comment:
X-Amz-Content-Sha256 can be set by some aws-sdks already to do server validation and it is supported. 

So we don't need to remove it - s3 docs are outdated :-)